src/ReverseProxy.Core/Service/Config/DynamicConfigBuilder.cs

```
@@ -47,40 +51,80 @@ public async Task<Result<DynamicConfigRoot>> BuildConfigAsync(IConfigErrorReport
             return Result.Success(config);
         }
 
+        public async Task<IDictionary<string, Backend>> GetBackendsAsync(IConfigErrorReporter errorReporter, CancellationToken cancellation)
+        {
+            var backends = await _backendsRepo.GetBackendsAsync(cancellation) ?? new Dictionary<string, Backend>(StringComparer.Ordinal);
+            var configuredBackends = new Dictionary<string, Backend>(StringComparer.Ordinal);
+            // The IBackendsRepo provides a fresh snapshot that we need to reconfigure each time.
+            foreach (var (id, backend) in backends)
+            {
+                try
+                {
+                    foreach (var filter in _filters)
+                    {
+                        await filter.ConfigureBackendAsync(id, backend, cancellation);
+                    }
+
+                    configuredBackends[id] = backend;
+                }
+                catch (Exception ex)
+                {
+                    errorReporter.ReportError(ConfigErrors.ConfigBuilderBackendException, id, "An exception was thrown from the configuration callbacks.", ex);
```

Does this swallow and log an error if a filter throws at startup? It seems better to let the exception bubble up to the caller if it's during server startup rather than config reload like Kestrel does.

---

IConfigErrorReporter is provided by the caller, not as a service. It's purpose is to collect all config errors before acting on them. The current implementation up in ProxyConfigLoader does log and swallow.